File: src/main/java/it/contrader/dto/InterestgroupDTO.java
package it.contrader.dto;

import it.contrader.model.Interest;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor

public class InterestgroupDTO {

	private Long id;

	private String interestgroup;
	
	private Interest interest;

}

File: Project1/DTO/InterestGroupDto.cs
using Project1.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Project1.DTO
{
	public class InterestGroupDto
	{
		public int? Id { get; set; }
		public string InterestGroupString { get; set; }
        public int InterestId { get; set; }
        public Interest Interest { get; set; }

        public InterestGroup ConvertTo()
        {
            return new InterestGroup
            {
                Id = this.Id,
                InterestGroupString = this.InterestGroupString,
                InterestId = this.InterestId,
                Interest = this.Interest
            };
        }
        public static InterestGroupDto ConvertFrom(InterestGroup interestGroup)
        {
            return new InterestGroupDto
            {
                Id = interestGroup.Id,
                InterestGroupString = interestGroup.InterestGroupString,
                InterestId = interestGroup.InterestId,
                Interest = interestGroup.Interest,
            };
        }
    }
}
